### PR TITLE
Update upstream DNS server capability descriptions

### PR DESCRIPTION
**Pull request by xanoni**
**By submitting this pull request, I confirm the following:** 
*please fill any appropriate checkboxes, e.g: [X]*

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [x] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

Please make sure you [Sign Off](https://github.com/pi-hole/pi-hole/wiki/How-to-signoff-your-commits.) all commits. Pi-hole enforces the [DCO](https://github.com/pi-hole/pi-hole/wiki/Contributing-to-the-project).

---
**What does this PR aim to accomplish?:**
*A detailed description, screenshots (if necessary), as well as links to any relevant GitHub issues*

Make it easier for users to select upstream DNS servers.

**How does this PR accomplish the above?:**
*A detailed description (such as a changelog) and screenshots (if necessary) of the implemented fix*

Mention that the below 3 upstream DNS support DNSSEC:
- Cloudflare (see https://developers.cloudflare.com/1.1.1.1/faq#how-does-1111-work-with-dnssec)
- DNS.WATCH (see https://dns.watch/index)
- Google (see https://developers.google.com/speed/public-dns/faq#dnssec)

Other providers and capabilities (e.g., ECS) were not checked.

**What documentation changes (if any) are needed to support this PR?:**
*A detailed list of any necessary changes*

None.

---
* You must follow the template instructions. Failure to do so will result in your pull request being closed.
* Please respect that Pi-hole is developed by volunteers, who can only reply in their spare time.

### CODE REVIEW

**Comment by yubiuser** [comment]
All return `ad` flag.

```
rockpi@rockpi-4b:~$ dig +dnssec sigok.verteiltesysteme.net @8.8.8.8

; <<>> DiG 9.11.5-P4-5.1+deb10u5-Debian <<>> +dnssec sigok.verteiltesysteme.net @8.8.8.8
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 24309
;; flags: qr rd ra ad; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags: do; udp: 512
;; QUESTION SECTION:
;sigok.verteiltesysteme.net.	IN	A

;; ANSWER SECTION:
sigok.verteiltesysteme.net. 60	IN	A	134.91.78.139
sigok.verteiltesysteme.net. 60	IN	RRSIG	A 5 3 60 20211230020002 20210930020002 30665 verteiltesysteme.net. JGnYujVvMvF3EXERDt8o4Et/4MJhzEnoaeK5rSq012ilkFbdre6OUd+B Uu1IB+30Z8WiTUPutmAt6DsqJXzapUxKBWLzgHPQ/4Y3BBTFzzZTykeX 4UgE2A+3LAMKOZWtvkgoAJwNma7UUoghKAl4o7ulendWpnbJ1vmjZcL1 VYY=

;; Query time: 60 msec
;; SERVER: 8.8.8.8#53(8.8.8.8)
;; WHEN: Mon Oct 04 08:03:13 CEST 2021
;; MSG SIZE  rcvd: 251

rockpi@rockpi-4b:~$ dig +dnssec sigok.verteiltesysteme.net @1.1.1.1

; <<>> DiG 9.11.5-P4-5.1+deb10u5-Debian <<>> +dnssec sigok.verteiltesysteme.net @1.1.1.1
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 15755
;; flags: qr rd ra ad; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags: do; udp: 1232
;; QUESTION SECTION:
;sigok.verteiltesysteme.net.	IN	A

;; ANSWER SECTION:
sigok.verteiltesysteme.net. 60	IN	A	134.91.78.139
sigok.verteiltesysteme.net. 60	IN	RRSIG	A 5 3 60 20211230020002 20210930020002 30665 verteiltesysteme.net. JGnYujVvMvF3EXERDt8o4Et/4MJhzEnoaeK5rSq012ilkFbdre6OUd+B Uu1IB+30Z8WiTUPutmAt6DsqJXzapUxKBWLzgHPQ/4Y3BBTFzzZTykeX 4UgE2A+3LAMKOZWtvkgoAJwNma7UUoghKAl4o7ulendWpnbJ1vmjZcL1 VYY=

;; Query time: 144 msec
;; SERVER: 1.1.1.1#53(1.1.1.1)
;; WHEN: Mon Oct 04 08:03:23 CEST 2021
;; MSG SIZE  rcvd: 251

rockpi@rockpi-4b:~$ dig +dnssec sigok.verteiltesysteme.net @84.200.69.80

; <<>> DiG 9.11.5-P4-5.1+deb10u5-Debian <<>> +dnssec sigok.verteiltesysteme.net @84.200.69.80
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 3011
;; flags: qr rd ra ad; QUERY: 1, ANSWER: 2, AUTHORITY: 3, ADDITIONAL: 7

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags: do; udp: 4096
;; QUESTION SECTION:
;sigok.verteiltesysteme.net.	IN	A

;; ANSWER SECTION:
sigok.verteiltesysteme.net. 60	IN	A	134.91.78.139
sigok.verteiltesysteme.net. 60	IN	RRSIG	A 5 3 60 20211230020002 20210930020002 30665 verteiltesysteme.net. JGnYujVvMvF3EXERDt8o4Et/4MJhzEnoaeK5rSq012ilkFbdre6OUd+B Uu1IB+30Z8WiTUPutmAt6DsqJXzapUxKBWLzgHPQ/4Y3BBTFzzZTykeX 4UgE2A+3LAMKOZWtvkgoAJwNma7UUoghKAl4o7ulendWpnbJ1vmjZcL1 VYY=

;; AUTHORITY SECTION:
verteiltesysteme.net.	3600	IN	NS	ns2.verteiltesysteme.net.
verteiltesysteme.net.	3600	IN	NS	ns1.verteiltesysteme.net.
verteiltesysteme.net.	3600	IN	RRSIG	NS 5 2 3600 20211230020002 20210930020002 30665 verteiltesysteme.net. u9h/A6eMbcfURlv+PfLI3Ocxa8TgF2lqIkaMbx2IJ4Zd/XbN2OhRH1s8 LuF0v+y/Uu2CMyrXYhfAKUHxwjsLyICz1GIOtPa+RCCqzW0VthPT2zpS 5WTt/RoHlgclFuJ+bCJ9nWw7N2sVV8URCjUUmnoc6o8t6erH9W3HcUlG 9fs=

;; ADDITIONAL SECTION:
ns1.verteiltesysteme.net. 3600	IN	A	134.91.78.139
ns2.verteiltesysteme.net. 3600	IN	A	134.91.78.141
ns1.verteiltesysteme.net. 3600	IN	AAAA	2001:638:501:8efc::139
ns1.verteiltesysteme.net. 3600	IN	RRSIG	A 5 3 3600 20211230020002 20210930020002 30665 verteiltesysteme.net. BRCYVTACUvTzHlofyC6td82bo24Tkop1hJu621k3lqJWQ5RVwHLNeQJ+ B69OADti2sjobq7BGnpORlo/wtvi7qvQhgY+OD+qhMze/ELwKNCUrOh4 6YMeWNjgMs23VRJq/LkRlcHpNsDEJkmI2A/eOpTi3dBF/+QspyNmtY4O 23k=
ns2.verteiltesysteme.net. 3600	IN	RRSIG	A 5 3 3600 20211230020002 20210930020002 30665 verteiltesysteme.net. k+J3Aof8JTBNOE2RWbhRGxf3Q5TEDgBy6SP/J/htjPnAgm3hvU/1WN6R eL2fDWAE0fSa5bObynpgLhDcMcRXEMGRLTV/KUU7XVGdNPTxHJXGAzQs E9m1tG0D80hbJC6mNEo5PuQ/zYFuos8Ei63V17uCYWfOtwDO6kukFMX0 sgM=
ns1.verteiltesysteme.net. 3600	IN	RRSIG	AAAA 5 3 3600 20211230020002 20210930020002 30665 verteiltesysteme.net. SCE/gqb78i096/szSqMnflZi717dHlUxX5ScFhEfdUK7eb1ZKVurZKqY ShWzF5qhIHRL2zd7EH5hU+S3MIDDXgX3HYfev0JeX1+lizG1Z5UFi/JO WnAWWH7r3T8NC+1ErwiNG/I1DIYydQV1MUlfU0x83hwa6r+sp1fUoNAu Z+c=

;; Query time: 43 msec
;; SERVER: 84.200.69.80#53(84.200.69.80)
;; WHEN: Mon Oct 04 08:03:37 CEST 2021
;; MSG SIZE  rcvd: 1067

```

**Review by yubiuser — APPROVED** [review]

**Comment by xanoni** [comment]
> All return `ad` flag.

Same is true for Level 3 and Comodo, by the way ... however, I just couldn't find a good confirmation from them when I searxed for it ...

**Comment by DL6ER** [comment]
> Same is true for Level 3 and Comodo, by the way ... however, I just couldn't find a good confirmation from them when I searxed for it ...

Yes, we should not add it when it is not documented as intended behavior. Otherwise, they may have it today and remove it again. tomorrow.

Also it might be even better to test for *rejection* instead, see Level 3 doesn't do DNSSEC here:
```
$ dig +dnssec sigfail.verteiltesysteme.net @4.2.2.1

; <<>> DiG 9.16.8-Ubuntu <<>> +dnssec sigfail.verteiltesysteme.net @4.2.2.1
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 10075
;; flags: qr rd ra; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags: do; udp: 8192
;; QUESTION SECTION:
;sigfail.verteiltesysteme.net.	IN	A

;; ANSWER SECTION:
sigfail.verteiltesysteme.net. 60 IN	A	134.91.78.139
sigfail.verteiltesysteme.net. 60 IN	RRSIG	A 5 3 60 20211230020002 20210930020002 30665 verteiltesysteme.net. //This+RRSIG+is+deliberately+broken///For+more+informati on+please+go+to/http+//www+verteiltesysteme+net///////// //////////////////////////////////////////////////////// //8=

;; Query time: 95 msec
;; SERVER: 4.2.2.1#53(4.2.2.1)
;; WHEN: Mo Okt 04 09:09:46 CEST 2021
;; MSG SIZE  rcvd: 253
```
Comodo does.

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -34,15 +34,15 @@ export PATH+=':/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin'
...
-Google (ECS);8.8.8.8;8.8.4.4;2001:4860:4860:0:0:0:0:8888;2001:4860:4860:0:0:0:0:8844
+Google (ECS, DNSSEC);8.8.8.8;8.8.4.4;2001:4860:4860:0:0:0:0:8888;2001:4860:4860:0:0:0:0:8844
 OpenDNS (ECS, DNSSEC);208.67.222.222;208.67.220.220;2620:119:35::35;2620:119:53::53
 Level3;4.2.2.1;4.2.2.2;;
 Comodo;8.26.56.26;8.20.247.20;;
-DNS.WATCH;84.200.69.80;84.200.70.40;2001:1608:10:25:0:0:1c04:b12f;2001:1608:10:25:0:0:9249:d69b
+DNS.WATCH (DNSSEC);84.200.69.80;84.200.70.40;2001:1608:10:25:0:0:1c04:b12f;2001:1608:10:25:0:0:9249:d69b
 Quad9 (filtered, DNSSEC);9.9.9.9;149.112.112.112;2620:fe::fe;2620:fe::9
 Quad9 (unfiltered, no DNSSEC);9.9.9.10;149.112.112.10;2620:fe::10;2620:fe::fe:10
 Quad9 (filtered + ECS);9.9.9.11;149.112.112.11;2620:fe::11;2620:fe::fe:11
```
Comment (MichaIng) [inline automated install/basic-install.sh:44]: The above third Cloudflare DNS supports DNSSEC as well: https://www.quad9.net/support/faq/#dnssec

Comment (xanoni) [reply automated install/basic-install.sh]: cool updated + rebased

**Pull request merged**